src/main.cpp

```diff
@@ -2540,16 +2631,35 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
+            if (!viewTempCache.HaveInputs(tx)) {
+                // ** In the case of IBD we have to check if the height has changed while we're checking inputs because this thread can continue for a
+                // short while after it has been interrupted and meanwhile the view of the UTXO could have changed after another thread has updated the tip.
+                // ** Additionally and more importantly in the event of a big block DDOS attack, by checking the chain height here, we ensure that
+                //    the thread that is validating the big block will immediately exit and finish up (while still keeping the block on disk in the
+                //    case of a reorg) as soon as the first block makes it through and wins the validation race.
+                if (fParallel)
+                {
+                    if (chainActive.Height() > nStartingHeight) {
```

Comment:
This is not for a re-org.   Imagine we have two competing threads.  Each is validating against the current chainactive tip to  get it's block through.  Now one of those has won the race and advanced the tip...the chain height will now be greater by 1 than when this thread began its validation. We could use chain work here also but it's the same thing at this point.